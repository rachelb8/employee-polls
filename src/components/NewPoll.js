import { useState } from "react";
import { connect } from "react-redux";
import { handleAddPoll } from "../actions/polls";
import { useNavigate } from "react-router-dom";
import { MDBInput, MDBBtn } from "mdb-react-ui-kit";
import AskLoginPage from "./AskLoginPage";

const NewPoll = ({ dispatch, authedUser }) => {
  const navigate = useNavigate();
  const [optionOne, setOptionOne] = useState("");
  const [optionTwo, setOptionTwo] = useState("");

  const handleChange = (e) => {
    const text = e.target.value;
    const option = e.target.id;

    option === "optionOne" ? setOptionOne(text) : setOptionTwo(text);
  };

  const handleClick = (e) => {
    e.preventDefault();

    dispatch(handleAddPoll(optionOne, optionTwo));
    navigate("/");
  };

  if (!authedUser) {
    return <AskLoginPage />;
  }

  return (
    <div className="text-center m-4 d-flex align-items-center justify-content-center">
      <div style={{ width: "48rem" }}>
        <h1>Would You Rather?</h1>
        <h3 className="text-muted">Create Your Own Poll</h3>
        <h4>First Option</h4>
        <MDBInput
          wrapperClass="mb-4"
          id="optionOne"
          label="Option One"
          type="text"
          onChange={handleChange}
        />
        <h4>Second Option</h4>
        <MDBInput
          wrapperClass="mb-4"
          id="optionTwo"
          label="Option Two"
          type="text"
          onChange={handleChange}
        />
        <MDBBtn
          onClick={handleClick}
          disabled={optionOne === "" || optionTwo === ""}
        >
          Submit
        </MDBBtn>
      </div>
    </div>
  );
};

function mapStateToProps({ authedUser }) {
  return {
    authedUser,
  };
}

export default connect(mapStateToProps)(NewPoll);
