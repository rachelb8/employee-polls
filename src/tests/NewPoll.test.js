import * as React from "react";
import { render, fireEvent } from "@testing-library/react";
import { Provider } from "react-redux";
import { createStore } from "redux";
import { BrowserRouter as Router } from "react-router-dom";
import middleware from "../middleware";
import reducer from "../reducers";
import NewPoll from "../components/NewPoll";

const store = createStore(reducer, middleware);

describe("NewPoll", () => {
  let component;
  let inputOptionOne;
  let inputOptionTwo;
  let submitButton;

  beforeEach(() => {
    component = render(
      <Provider store={store}>
        <Router>
          <NewPoll />
        </Router>
      </Provider>
    );

    inputOptionOne = component.getByTestId("option-one-input");
    inputOptionTwo = component.getByTestId("option-two-input");
    submitButton = component.getByTestId("submit-button");
  })

  it("matches the snapshot", () => {
    expect(component).toMatchSnapshot();
  });

  it("will enable the submit button when submitting a new poll with both options", async () => {
    fireEvent.change(inputOptionOne, { target: { value: "option one test" } });
    fireEvent.change(inputOptionTwo, { target: { value: "option two test" } });

    expect(submitButton).toBeEnabled();
  });

  it('will disable the submit button when submitting a new poll without both options', () => {
    fireEvent.change(inputOptionOne, { target: { value: "option one test" } });
    fireEvent.change(inputOptionTwo, { target: { value: "" } });

    expect(submitButton).toBeDisabled();
  });
});
