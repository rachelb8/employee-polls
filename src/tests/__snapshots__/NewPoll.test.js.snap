// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewPoll matches the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="text-center m-4 d-flex align-items-center justify-content-center"
      >
        <div
          style="width: 48rem;"
        >
          <h1>
            Would You Rather?
          </h1>
          <h3
            class="text-muted"
          >
            Create Your Own Poll
          </h3>
          <h4>
            First Option
          </h4>
          <div
            class="form-outline mb-4"
          >
            <input
              class="form-control"
              data-testid="option-one-input"
              id="optionOne"
              type="text"
              value=""
            />
            <label
              class="form-label"
              for="optionOne"
            >
              Option One
            </label>
            <div
              class="form-notch"
            >
              <div
                class="form-notch-leading"
              />
              <div
                class="form-notch-middle"
                style="width: 0px;"
              />
              <div
                class="form-notch-trailing"
              />
            </div>
          </div>
          <h4>
            Second Option
          </h4>
          <div
            class="form-outline mb-4"
          >
            <input
              class="form-control"
              data-testid="option-two-input"
              id="optionTwo"
              type="text"
              value=""
            />
            <label
              class="form-label"
              for="optionTwo"
            >
              Option Two
            </label>
            <div
              class="form-notch"
            >
              <div
                class="form-notch-leading"
              />
              <div
                class="form-notch-middle"
                style="width: 0px;"
              />
              <div
                class="form-notch-trailing"
              />
            </div>
          </div>
          <button
            class="ripple ripple-surface btn btn-primary"
            data-testid="submit-button"
            disabled=""
            role="button"
          >
            Submit
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="text-center m-4 d-flex align-items-center justify-content-center"
    >
      <div
        style="width: 48rem;"
      >
        <h1>
          Would You Rather?
        </h1>
        <h3
          class="text-muted"
        >
          Create Your Own Poll
        </h3>
        <h4>
          First Option
        </h4>
        <div
          class="form-outline mb-4"
        >
          <input
            class="form-control"
            data-testid="option-one-input"
            id="optionOne"
            type="text"
            value=""
          />
          <label
            class="form-label"
            for="optionOne"
          >
            Option One
          </label>
          <div
            class="form-notch"
          >
            <div
              class="form-notch-leading"
            />
            <div
              class="form-notch-middle"
              style="width: 0px;"
            />
            <div
              class="form-notch-trailing"
            />
          </div>
        </div>
        <h4>
          Second Option
        </h4>
        <div
          class="form-outline mb-4"
        >
          <input
            class="form-control"
            data-testid="option-two-input"
            id="optionTwo"
            type="text"
            value=""
          />
          <label
            class="form-label"
            for="optionTwo"
          >
            Option Two
          </label>
          <div
            class="form-notch"
          >
            <div
              class="form-notch-leading"
            />
            <div
              class="form-notch-middle"
              style="width: 0px;"
            />
            <div
              class="form-notch-trailing"
            />
          </div>
        </div>
        <button
          class="ripple ripple-surface btn btn-primary"
          data-testid="submit-button"
          disabled=""
          role="button"
        >
          Submit
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
